src/ORM/ArrayList.php

```diff
@@ -175,23 +175,29 @@ public function toNestedArray()
         return $result;
     }
 
-    /**
-     * Get a sub-range of this dataobjectset as an array
-     * Pass null to "remove the limit" - this is for consistency with DataList::limit(null) which itself will
-     * call SQLSelect::setLimit(null)
-     *
-     * @param int|null $length
-     * @param int $offset
-     * @return static
-     */
-    public function limit($length, $offset = 0)
+    public function limit(?int $length, int $offset = 0): static
     {
-        if (!$length) {
-            $length = count($this->items ?? []);
+        if ($length === null) {
+            // If we unset the limit, we set the length to the size of the list. We still want the offset to be picked up
+            $length = sizeof($this->items);
```

Comment:
What's the advantage of `sizeof` over `count`? `count` seems to be the more standard way to get the number of items in a `Countable` or `array`.

Reply:
Apparently it's exactly the same thing. https://www.php.net/manual/en/function.sizeof.php

For some reason, I've always defaulted to `sizeof`. Don't have any preference either way.